smslibrary/src/main/java/com/eis/smslibrary/SMSReceivedBroadcastReceiver.java

```diff
@@ -30,30 +33,36 @@ public void onReceive(Context context, Intent intent) {
         if (extras == null) {
             return;
         }
-
-        Object[] smsExtras = (Object[]) extras.get("pdus");
+        Object[] pdus = (Object[]) extras.get("pdus");
+        if (pdus == null) return;
         String format = (String) extras.get("format");
+        Map<String, String> peerMessageMap = new HashMap<>(pdus.length);
 
-        if(smsExtras == null) //could be null
-            return;
-
-        for (Object smsData : smsExtras) {
-            SmsMessage receivedSMS = createMessageFromPdu(smsData, format);
-            String smsContent = receivedSMS.getMessageBody();
-            String phoneNumber = receivedSMS.getOriginatingAddress();
-
-            if (phoneNumber == null) //could be null
-                return;
-
-            SMSMessage parsedMessage = SMSMessageHandler.getInstance().parseMessage(phoneNumber, smsContent);
-            if (parsedMessage != null) {
-                callApplicationService(context, parsedMessage);
+        /* There can be multiple SMSes from multiple senders. We'll concatenate multiple SMSes from
+         * the same sender in one message.
+         */
+        for (Object pdu : pdus) {
+            SmsMessage message = createMessageFromPdu(pdu, format);
+            String phoneNumber = message.getOriginatingAddress();
+            if (phoneNumber == null) phoneNumber = "";
+            if (!peerMessageMap.containsKey(phoneNumber)) {
+                peerMessageMap.put(phoneNumber, message.getMessageBody());
+            } else {
+                String previousParts = peerMessageMap.get(phoneNumber);
+                String updatedMessage = previousParts + message.getMessageBody();
+                peerMessageMap.put(phoneNumber, updatedMessage);
             }
         }
+        for (String peer : peerMessageMap.keySet()) {
+            SMSMessage parsedMessage =
+                    SMSMessageHandler.getInstance().parseMessage(peer, peerMessageMap.get(peer));
```

Comment:
Maybe add a a few lines to explain what parseMessage() does. I know I should check the method, but it's just a reminder for anyone who looks at this file only. You just obtain the string message from getMessageBody() so why do you need to parse it again?

Reply:
It's not a method i wrote, it's used to check for the special character used to identify messages sent from apps using this library. I left it because it was used in the receiver for single messages too, however I know that using this approach to identify messages introduces some bugs which i pointed out in https://github.com/Cogno-IDU/smslibrary/pull/17#discussion_r358004361

I'll add a comment as you said

Reply:
OK